Userland/Libraries/LibJS/Runtime/ObjectPrototype.cpp

```diff
@@ -147,6 +150,70 @@ JS_DEFINE_NATIVE_FUNCTION(ObjectPrototype::is_prototype_of)
     }
 }
 
+// B.2.2.2 Object.prototype.__defineGetter__ ( P, getter ), https://tc39.es/ecma262/#sec-object.prototype.__defineGetter__
+JS_DEFINE_NATIVE_FUNCTION(ObjectPrototype::define_getter)
+{
+    auto object = vm.this_value(global_object).to_object(global_object);
+    if (vm.exception())
+        return {};
+
+    auto getter = vm.argument(1);
+    if (!getter.is_function()) {
+        vm.throw_exception<TypeError>(global_object, ErrorType::NotAFunction, getter.to_string_without_side_effects());
+        return {};
+    }
+
+    auto key = vm.argument(0).to_property_key(global_object);
+    if (vm.exception())
+        return {};
+
+    auto descriptor = Object::create(global_object, global_object.object_prototype());
+    descriptor->define_property(vm.names.get, getter);
+    descriptor->define_property(vm.names.enumerable, Value(true));
+    descriptor->define_property(vm.names.configurable, Value(true));
```

Comment:
Having to allocate a new object just for this shows how terrible our home-grown variant of the spec's API is, but that's another yak for another day.